### ISSUE TITLE
PEP 695: Type Parameter Syntax

### ISSUE THREAD

**Issue by Yamahari** [pylint-dev/pylint#9726]
### Bug description

```python
# pylint: disable=C0304,C0114,C0115,R0903
class GenericClass[T]:
    def __init__(self, x: T) -> None:
        pass
```

### Command used

```shell
pylint src\testtyping\typing.py
```

### Pylint output

```python
************* Module testtyping.typing
src\testtyping\typing.py:3:26: E0602: Undefined variable 'T' (undefined-variable)
```

### Expected behavior

No Error

### Pylint version

```shell
pylint 3.2.3
astroid 3.2.2
Python 3.12.4 (tags/v3.12.4:8e8a4ba, Jun  6 2024, 19:30:16) [MSC v.1940 64 bit (AMD64)]
```

### OS / Environment

Windows 10 Enterprise 22H2
